privacyidea/lib/crypto.py

```diff
@@ -403,25 +413,34 @@ def aes_encrypt_b64(key, data):
     :return: base64 encrypted output, containing IV and encrypted data
     :rtype: str
     """
+    # pad data
+    block_size = algorithms.AES.block_size // 8
+    num_pad = block_size - (len(data) % block_size)
+    data = data + six.int2byte(num_pad) * num_pad
```

Comment:
Hmm, this would be PKCS7 padding? In this case, we could also just use the [padders from cryptography](https://cryptography.io/en/latest/hazmat/primitives/padding/#cryptography.hazmat.primitives.padding.PKCS7). What do you think?

Reply:
Oh, just noticed: Don't we have a problem if the input size is divisible by 16, because then we add a padding block of null bytes when encrypting, but we don't remove it when decrypting?
... but if this is the case, the old code had the same problem.

Reply:
Exactly. The problem is the unpadding. The cryptography unpadder would error out on an incorrectly padded input.
So we would need some logic here to check both versions or make a hard cut.

Reply:
Sorry, i didn't read carefully.
Actually we are using correct PKCS7 padding here and i should use the cryptography padders here. I'll fix that.
If the input size is divisible by 16, we would have `num_pad = 16` and thus add 16 times `b'\x16'` to the data, i.e. a whole block. And this should be correctly removed by the unpadder.

Reply:
I misread too :-) You're right, I think we do correct PKCS7 padding here.

Reply:
fixed in f2e066a6

Reply:
thanks!